modules/standard/IO.chpl

```diff
@@ -8095,39 +8097,93 @@ private proc readBytesOrString(ch: fileReader, ref out_var: ?t, len: int(64)) :
 
```

Comment:
I would like the calculation of `uselen` to be moved to a helper function so that `uselen` can be `const`. (Same for `guessReadSize below.) Add a comment that `uselen` is the limit on the read size based on the `len` argument and the channel size.

Also, in the above, I suggest changing the test `len == -1` to `len < 0` to prevent negative `uselen`.